pilot/pkg/proxy/envoy/v2/discovery.go

```diff
@@ -300,23 +307,34 @@ func debounce(ch chan *model.PushRequest, stopCh <-chan struct{}, pushFn func(re
 	}
 
 	pushWorker := func() {
-		eventDelay := time.Since(startDebounce)
+		debounceDelay := time.Since(startDebounce)
 		quietTime := time.Since(lastConfigUpdateTime)
-		// it has been too long or quiet enough
-		if eventDelay >= debounceMax || quietTime >= debounceAfter {
+		// it has been too long or quiet enough - push the current requests.
+		if debounceDelay >= debounceMax || quietTime >= debounceBackoff {
```

Comment:
Not sure this is right. If we do exponential back off and let’s say we are currently at 8s, what this will mean is that after the last apply from a large yaml file, there will be 8s of delay to push configs.

Reply:
Do you think we should stick with the old approach while determining the quiet time i.e. if there are no events in the `debounceAfter` (100ms) - we just push the config? I was debating about that  - If that is better, I can change it. or any other better ideas?